test/e2e/predicates/predicates_test.go

```diff
@@ -251,31 +243,25 @@ var _ = Describe("Predicates", func() {
 			},
 		}
 
-		var success bool
 		initPod, podErr := k8s.InitTestPod(conf)
 		Ω(podErr).NotTo(HaveOccurred())
-		success, err = kClient.WaitForSchedulerAfterAction(kClient.CreateTestPodAction(initPod, ns), ns, podName, false)
-		Ω(err).NotTo(HaveOccurred())
-		Ω(success).Should(BeTrue())
+		_, podErr = kClient.CreatePod(initPod, ns)
+		Ω(podErr).NotTo(HaveOccurred())
 
 		By(fmt.Sprintf("Verify pod:%s is in pending state", podName))
-		err = kClient.WaitForPodPending(ns, podName, time.Duration(60)*time.Second)
-		Ω(err).NotTo(HaveOccurred())
+		podErr = kClient.WaitForPodPending(ns, podName, time.Duration(60)*time.Second)
+		Ω(podErr).NotTo(HaveOccurred())
 
-		By("Verify the Yunikorn events: failed scheduling")
-		var events *v1.EventList
-		events, err = kClient.GetEvents(ns)
-		Ω(err).NotTo(HaveOccurred())
-		Ω(len(events.Items)).NotTo(BeZero(), "Events cant be empty")
-		for _, event := range events.Items {
-			framework.Logf("Event source is : %s", event.Source.Component)
-			if event.Source.Component == configmanager.SchedulerName && event.Type == WARNING {
-				framework.Logf("Yunikorn Event Reason: %s", event.Reason)
-				Ω(event.Reason).Should(Equal("FailedScheduling"), "Event reason mismatch")
-				framework.Logf("Yunikorn Event Message: %s", event.Message)
-				Ω(event.Message).Should(MatchRegexp("predicate is not satisfied.*didn't match Pod's node affinity"), "Event message mismatch")
-			}
-		}
+		By("Verify the YuniKorn request failed scheduling")
+
+		podErr = restClient.WaitForAllocationLog("default", "root."+ns, initPod.ObjectMeta.Labels["applicationId"], podName, 60)
+		Ω(podErr).NotTo(HaveOccurred())
+		log, podErr := restClient.GetAllocationLog("default", "root."+ns, initPod.ObjectMeta.Labels["applicationId"], podName)
+		Ω(podErr).NotTo(HaveOccurred())
+		Ω(len(log)).NotTo(BeZero(), "Log can't be empty")
+
+		Ω(log).Should(ContainElement(HaveKeyWithValue("message", MatchRegexp(".*didn't match Pod's node affinity"))),
```

Comment:
Can we check by matching with earlier message? I am assuming allocation log contains same message.

Reply:
These match against the message the K8s predicates generate. We purposely keep them generic so as to be more flexible when new K8s versions may change the wording slightly. There's not really a risk of false positives, as only messages from predicates are logged to the allocation log.